File: php/features.php
<?php
declare(strict_types=1);

// KanjiData SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class KanjiDataFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new KanjiDataBaseFeature();
            case "test":
                return new KanjiDataTestFeature();
            default:
                return new KanjiDataBaseFeature();
        }
    }
}

File: py/features.py
# KanjiData SDK feature factory

from feature.base_feature import KanjiDataBaseFeature
from feature.test_feature import KanjiDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: KanjiDataBaseFeature(),
        "test": lambda: KanjiDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
